Code/src/main/java/nl/utwente/group10/ui/handlers/AnchorHandler.java

```diff
@@ -0,0 +1,77 @@
+package nl.utwente.group10.ui.handlers;
+
+import javafx.event.EventHandler;
+import javafx.scene.Node;
+import javafx.scene.input.InputEvent;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.input.TouchEvent;
+import javafx.scene.input.TouchPoint;
+import nl.utwente.group10.ui.components.ConnectionAnchor;
+
+	public class AnchorHandler implements EventHandler<InputEvent> {
+
+		private ConnectionCreationManager manager;
+		private ConnectionAnchor anchor;
+
+		public AnchorHandler(ConnectionCreationManager manager,
+				ConnectionAnchor anchor) {
+			this.manager = manager;
+			this.anchor = anchor;
+
+			anchor.addEventFilter(MouseEvent.MOUSE_PRESSED, this);
+			anchor.addEventFilter(MouseEvent.MOUSE_DRAGGED, this);
+			anchor.addEventFilter(MouseEvent.MOUSE_RELEASED, this);
+			
+			anchor.addEventFilter(TouchEvent.TOUCH_PRESSED, this);
+			anchor.addEventFilter(TouchEvent.TOUCH_MOVED, this);
+			anchor.addEventFilter(TouchEvent.TOUCH_RELEASED, this);
+		}
+
+		@Override
+		public void handle(InputEvent event) {
+			Node pickResult = null;
+			int inputId = -1;
+			double x = 0;
+			double y = 0;
+
+			if (event instanceof MouseEvent
+					&& !((MouseEvent) event).isSynthesized()) {
+				MouseEvent mEvent = ((MouseEvent) event);
+				pickResult = mEvent.getPickResult().getIntersectedNode();
+				inputId = ConnectionCreationManager.MOUSE_ID;
+				x = mEvent.getSceneX();
+				y = mEvent.getSceneY();
+			} else if (event instanceof TouchEvent) {
+				TouchPoint tp = ((TouchEvent) event).getTouchPoint();
+				pickResult = tp.getPickResult().getIntersectedNode();
+				inputId = tp.getId();
+				x = tp.getSceneX();
+				y = tp.getSceneY();
+			}
+
+			if (pickResult != null && inputId >= 0) {
+				if (event.getEventType().equals(MouseEvent.MOUSE_PRESSED)
```

Comment:
Deze if-boom kan misschien wel wat netter. Je kunt in ieder geval event.getEventType() een naam geven, dan hoef je dat niet steeds te herhalen. Ook zou je de bodies van wat er moet gebeuren voor elk event naar eigen methoden kunnen trekken. Als je dan vervolgens per event een eigen methode hebt kun je overwegen die methode meteen aan addEventFilter mee te geven, dan hoef je de if helemaal niet meer te doen.